edd-license-free-download.php

```diff
@@ -276,28 +276,10 @@ public static function validate_license( $license_key ) {
 	 *
 	 * @param string $license_key
 	 *
-	 * @return array
+	 * @return int
 	 */
 	public static function get_license_product_ids( $license_key ) {
-
-		global $wpdb;
-
-		// The edd_license post ID of the license key
-		$license_post_id = $wpdb->get_var(
-		$wpdb->prepare(
-		"SELECT post_id FROM $wpdb->postmeta WHERE meta_key = '_edd_sl_key' AND meta_value = %s", $license_key )
-		);
-
-		$payment_id = get_post_meta( $license_post_id, '_edd_sl_payment_id', true );
-
-		$get_product_id_for_payment = edd_get_payment_meta_cart_details( $payment_id );
-
-		$products_ids_for_payment = array();
-		foreach ( $get_product_id_for_payment as $key => $products ) {
-			$products_ids_for_payment[] = $products['id'];
-		}
-
-		return $products_ids_for_payment;
+		return edd_software_licensing()->get_download_id_by_license( $license_key );

```

Comment:
Two things:

1. I understand this used to return an array. Could that have actually contained more than one product ID? In short I'm trying to understand if/how this could have returned more than one value in the past, but now it's definitely only one?
2. Regardless of the answer to the first question, I'm concerned about changing this return type when it's a public method. It will cause issues for anyone using it.

Reply:
As I read the code in `master`, I think it was getting the payment ID, and then getting _all_ products associated with that payment? So it's possible it could have returned more than one value, if multiple items were in the cart? (It looks like it's getting the cart?) However, that doesn't seem like a logical way of getting the information needed.

I should be able to cast the new returned value to an array, and just use `array_intersect` for the validation. Would that be acceptable?